### PR TITLE
Member fixes

### LINKED ISSUE #10313: members plugin should not be doing fancy hook for tabs and page handlers
Better it be regular/normal filter menu manipulation and resource files per page....

### LINKED ISSUE #11298: Members search results page lacks filter tabs

### LINKED ISSUE #11306: master: no online users
Hey at least they don't fatal.

### PR DESCRIPTION
**Pull request by jdalsem**
- feat(members): use filter menu and resource views for the pages fixes #10313
- feat(members): added members tabs to member search page fixes #11298
- chore(users): correctly update last_action for ElggUser  fixes #11306

### CODE REVIEW

**Comment by hypeJunction** [comment]
Finally! Thank you.

File: mod/members/start.php [hunk 1]
```diff
@@ -15,181 +11,66 @@
...
-function members_list_online($hook, $type, $returnvalue, $params) {
-	if ($returnvalue !== null) {
-		return;
-	}
-	return get_online_users();
-}
-
-/**
- * Returns content for the "alphabetical" page
+ * @param \Elgg\Hook $hook 'register', 'menu:filter'
```
Comment (jeabakker) [inline mod/members/start.php]: 'menu:filter:members' NOT 'menu:filter'

File: mod/members/start.php [hunk 2]
```diff
@@ -15,181 +11,66 @@
...
-		return;
-	}
-	
-	$options = elgg_extract('options', $params);
-
-	$options['order_by_metadata'][] = [
-		'name' => 'name',
-		'direction' => 'ASC',
+function members_register_filter_menu(\Elgg\Hook $hook) {
+	$result = [
```
Comment (jeabakker) [inline mod/members/start.php]: why not respect the return value of the hook, and add the items you want?

**Review by jeabakker — DISMISSED** [review]

File: mod/members/start.php (hunk 1 above)
Comment (hypeJunction) [reply mod/members/start.php]: Use `@elgg_plugin_hook register menu:filter:members`

Comment (jdalsem) [reply mod/members/start.php]: i see you using that... but the usage of `@elgg_plugin_hook` is nowhere described... i do not even know its effect...

File: mod/members/start.php (hunk 2 above)
Comment (jdalsem) [reply mod/members/start.php]: changed

File: mod/members/start.php (hunk 1 above)
Comment (jdalsem) [reply mod/members/start.php]: changed it nonetheless

Comment (hypeJunction) [reply mod/members/start.php]: @mrclay started using it, but it makes sense, we can then detect them with annotation parsers.

**Pull request merged**